woocommerce/woocommerce
# Add Launch Newsletter panel for the coming soon template

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52444  .

This PR adds `Launch Newsletter` panel for the coming soon template when `coming-soon-newsletter-template` feature flag is enabled.

Note 1: I wan't able to build wc admin with `@wordpress/editor` package installed. For now, I'm using globally available [wp.editor](https://github.com/woocommerce/woocommerce/compare/add/52444-newsletter-extension-hint-panel-mailpoet2?expand=1#diff-a12d162f589f6094b61cdef10a0f03d7ed3fc143d087b7fb43a3e7612990cca8R26). It felt a hack at first, but I wonder if we can just use this..as it's always available when the Gutenberg is loaded 🤔 

Note 2: When MailPoet isn't installed, I'm linking `Set up Mailpoet` link to the plugin's [detail page](https://github.com/woocommerce/woocommerce/compare/add/52444-newsletter-extension-hint-panel-mailpoet2?expand=1#diff-a12d162f589f6094b61cdef10a0f03d7ed3fc143d087b7fb43a3e7612990cca8R55). Do we want to link to the plugin search result page instead?

Note 3: The panel renders before the Content panel, and I haven't found a way to render it after the Content panel. It seems that WordPress/gutenberg#18064 is still a work in progress.

![Screenshot 2024-11-07 at 11 10 46 AM](https://github.com/user-attachments/assets/136c263a-afa8-413c-a4e0-6a140b674a74)




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site this this branch.
2. Complete the core profiler and enable `coming-soon-newsletter-template` feature flag.
3. Make sure to use a block theme.
4. Navigate to `Appearance -> Editor -> Templates` and search for 'coming soon'
5. Click on `Page: Coming soon` template.
6. Confirm `Launch Newsletter` panel is shown.
7. Click on `Set up Mailpoet` link and you should be redirected to MialPoet plugin page to install. 
8. Once MailPoet is installed and activated, go back to the template and click `Set up Mailpoet` again.
9. You should be redirected to MailPoet config page.
12. Disable `coming-soon-newsletter-template` feature flag and go back to the template again.
13. `Launch Newsletter` panel should not be shown.
14. Enable the feature flag again and run `update wp_mailpoet_settings set value='a:2:{s:16:"mailpoet_api_key";s:4:"test";s:22:"mailpoet_api_key_state";a:4:{s:5:"state";s:5:"valid";s:18:"access_restriction";N;s:4:"data";N;s:4:"code";i:200;}}' where name='mta';` query. The query tricks MailPoet into thinking it is connected.
15. Go back to the template and the panel should not be shown.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add Launch Newsletter panel for the coming soon template.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
